src/queries/queries.js

```diff
@@ -100,6 +100,41 @@ export const buildContactsQueryByUpdatedAtGT = date => ({
     .limitBy(1000)
 })
 
+export const buildContactsQueryByEmailAdressOrPhoneNumber = (
+  addresses,
+  numbers
+) => {
+  return {
+    definition: Q(DOCTYPE_CONTACTS)
+      .where({
+        _id: {
+          $gt: null
```

Comment:
Cette requête cherche à trouver les contacts qui ont un email ou un téléphone qui matchent une liste donnée.
Pour faire ça, il faut  déjà s'assurer que les champs `email` et `phone` existent dans un partial index (c'était pas le cas précédememnt @cballevre ?).
Ensuite, il y a besoin d'évaluer ce `$or`. On a trouvé un bug dans cozy-client (je dois le documenter) qui fait qu'on va chercher un index sur le `$or`, d'où le besoin d'avoir un prédicat sur l'id à la place, et de l'indexer. Quand le bug côté cozy-client sera corrigé, il n'y aura pas besoin de cet _id

Reply:
@paultranvan They was into two separate queries with `where` and the `$or` was done by javascript with loadash functions.

Reply:
@JF-Cozy This query is used during import to search if the contact already exists. It's match on email address or phone numbers. If there is a match, the contact is updated instead of creating.